rest/route/host_spawn.go

```diff
@@ -876,36 +888,52 @@ func (h *modifyVolumeHandler) Run(ctx context.Context) gimlet.Responder {
 				Message:    err.Error(),
 			})
 		}
-		mgrOpts := cloud.ManagerOpts{
-			Provider: evergreen.ProviderNameEc2OnDemand,
-			Region:   cloud.AztoRegion(volume.AvailabilityZone),
+	}
+
+	if !utility.IsZeroTime(h.opts.Expiration) {
```

Comment:
Is all of this validation the same as for host expirations? Can this be made into a helper function excepting Expiration and NoExpiration that can be called for each of them?

Reply:
I added validation to HostModify to check the user isn't extending the expiration and setting no expiration.
I'm not sure about adding a helper function.

Reply:
Well my thought was that modular code removes error with keeping the validation the same and generally is easier to read, but I defer to whatever design you prefer, as this is already a large ticket

Reply:
I'm open to it, but I don't see other aspects of the validation which are common.